### DISCUSSION TITLE
ForgeUI Vue3 Migration Path

### DISCUSSION THREAD

**Discussion by TomSmith27** [3Squared/ForgeUI#110] [Ideas]
Vue 3 is now the default version of Vue, most features like the composition api have being back ported to vue2.7 which is what we use. However more and more libraries will soon stop supporting Vue2.7 so we need a migration strategy

**Comment by Polymeta** [comment]
Well, our biggest block is bootstrap-vue, isnt it? Whats the status on that? Last I heard it was in early compat stage. Based on this comment https://github.com/bootstrap-vue/bootstrap-vue/issues/6872#issuecomment-1439025813 - it looks like there might be a movement towards a new repo, which is here https://github.com/bootstrap-vue/bootstrap-vue-next - but thats still in early alpha.

I feel like it's hard for us to move forward to vue3 with the elephant in the room that is bootstrap-vue

**Reply by TomSmith27** [reply]
Yep bootstrap-vue is the biggest blocker, And whilst they are making progress they aren't filling me with confidence as it only appears to be a couple of people working on it. And it could easily just stop being worked on again.

I think we need to consider some of the following options.

1. Remaining on Vue 2.7 until when/if bootstrap-vue is updated.
2. Consider creating our own bootstrap vue implementation using bootstrap 5 and wrapping components ourselves
3. Use a new Technology

      1. Vuetify
     2. PrimeVue
     3. Tailwind
     4. Tailwind / Headless / Flowbite

**Reply by Polymeta** [reply]
Yeah, that's of the issue with open source. And yeah, I agree we only have those options left, but we might have to consider voting/pushing for option 2/3 if we chose to do so, however, the further down the options we go, the more time consuming and  complex the maintenance overhead gets.

Option is kind of the default, as we do that at the moment really. So it's more about "Do we want to stop doing option 1 and push for 2 or 3?", which might be hard to sell due to above mentioned effort and time this would need.

I'm not opposed to trying out new frameworks, but then we also have to ensure we don't lose functionalities that are currently being used. Additionally, at least from a user POV, we also have to consider the amount of time/effort that would go into styling these frameworks to match what we have now, as some users might not want to have to change their page designs around when we do make a ForgeUI update of this size.
